### ISSUE TITLE
Can’t run from disk, or make RancherOS persistent in any other method

### ISSUE THREAD

**Issue by efiweiss** [rancher/os#1946]
**RancherOS Version: (ros os version)**
0.8.0

**Where are you running RancherOS? (docker-machine, AWS, GCE, baremetal, etc.)**
VMware workstation 12 player (running on Win7)
___

Hi,

I've been trying to create a RancherOS VM on my laptop for almost a week now, but with no success :frowning:

I carefully followed the instructions here:
http://rancher.com/docs/os/running-rancheros/server/install-to-disk/
Also tried this walk-through: https://www.youtube.com/watch?v=_Od8Z3iv54M
And read many many other threads here.

The bottom line is that it seems the “cloud-config.yml” file I create is totally ignored, and the RancherOS is only restarting and running from Memory.
I can tell that based on many things, to list only 2:
* Ssh keys (I ingested in the cloud-config.yml) are not recognized.
* No partition exists on /dev/sda (it’s not mounted).

Any ideas??
Thanks!

**Comment by efiweiss** [comment]
I now noticed this error during startup:

![image](https://user-images.githubusercontent.com/19686061/27518694-57c2dff2-59ee-11e7-868a-b468e3e4e714.png)

**Comment by SvenDowideit** [comment]
I use VMWare (ESXi atm) quite  bit.

Can you please give us more specific details?

- What settings are you using in VMWare to create the VM
- What specific cmdline commands are you using, and what are their outputs
- how are you then rebooting & removing the initial boot media

**Comment by efiweiss** [comment]
thanks for your response.
I'm using mostly the default settings to create the VM (changing only CPU and HD size - tried different sizes with no impact on the result).

as for the command line - I'm using exactly the line in the link I mentioned above:
`sudo ros install -c cloud-config.yml -d /dev/sda`

there is not error in the output, the machine then reboots (as expected).

I don't know what you mean by removing the media. I'm using ISO file (on the local HD) to initially run the VM.

**Comment by SvenDowideit** [comment]
so it sounds like when it reboots, the VM is booting from the ISO again - you need to remove it from the VM's cfg. (yup, this happens to me too)

**Comment by efiweiss** [comment]
I tried to remove the "CD/DVD Drive" from the VM settings BEFORE executing the install.
same result :(

**Comment by gizmotronic** [comment]
@efiweiss since you've run the install process it should already be ready to go.

1. Boot the VM.
2. Disconnect the ISO image from the CD/DVD drive. VMware will warn you that it's in use; tell it you want to disconnect anyway.
3. Power off the VM. Since it's running from RAM there is no risk of damaging the file system on disk.
4. Power on the VM.

Et voilà – RancherOS should now boot using your initial cloud-config file, and future configuration changes should persist.

There are other ways to get to this same result, but my experience with RancherOS on VMware is that this way produces the fewest complications over the long run.
